### DISCUSSION TITLE
samples/subsys/usb/console zephyr 3.5.0 mimxrt1060_evkb immediate kernel fault

### DISCUSSION THREAD

**Discussion by asteriskSF** [zephyrproject-rtos/zephyr#69271] [Q&A] [answered]
I am testing USB on a MIMXRT1060_EVKB board.  I copied the sample/subsys/usb/console, built and flashed.  When debugging I find that usb_dc_attach returns with -EIO  here

	dev_state.dev_struct.controllerInterface = &mcux_usb_iface;
	status = dev_state.dev_struct.controllerInterface->deviceInit(CONTROLLER_ID,
						&dev_state.dev_struct,
						&dev_state.dev_struct.controllerHandle);
	if (kStatus_USB_Success != status) {
		return -EIO;
	}

Is USB supported on NXP platforms in zephyr 3.5.0 ? 
Do I need some additional configuration to make this sample work on this EVK? @

**Comment by asteriskSF** [comment, marked as answer]
After I re-built with --pristine the usb seems to be working now.  No idea what was going on earlier.

**Comment by asteriskSF** [comment]
BTW -- it appears the issue was building/flashing from MCUXpresso extension in VSCode.  I am only able to get USB to work when I run west build and west flash from the command line.

**Reply by DerekSnell** [reply]
Hi @asteriskSF ,
Sorry for the delay.  The MCUXpresso extension for VS Code did have some issues building samples from the Zephyr repo, and these were resolved in the latest extension v1.6.47.  I suspect that was the reason you could build this sample from CLI, but not with VS Code.  I just tested the VS Code extension v1.6.47 with Zephyr v3.6.0, and this USB Console sample works as expected on the MIMXRT1060-EVKB board.

Please note, with Zephyr's recent change to Hardware Model v2, the VS Code extension does not yet work with v2.  NXP is updating the extension to support v2, and this [Issue](https://github.com/nxp-mcuxpresso/vscode-for-mcux/issues/34) tracks the progress.  Until then, you can use VS Code with Zephyr v3.6.0 and the v1 model.

Best regards

**Reply by asteriskSF** [reply]
I have it all working reasonable well now, building from VSCode does work.  I'm not sure of the root cause.  Also I briefly tried 3.6.0 but realized the hardware model 2 and new experimental USB were too risky for the short term project I am focusing on..
